src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -279,7 +279,10 @@ public static String parsePaymentValue(String paymentValue) throws ParseExceptio
         requireNonNull(paymentValue);
         String trimmedPayment = paymentValue.trim();
         if (!Payment.isValidPayment(trimmedPayment)) {
-            throw new ParseException(Payment.MESSAGE_CONSTRAINTS);
+            if (!Payment.isPaymentWithAnyDecimals(trimmedPayment)) {
```

Comment:
Hmm it does seem a little too nested here with 2 if-statements, maybe you could consider splitting it up into 2 if blocks for better SLAP?

Reply:
Might also be possible to lump it together since its for the same parameter

Reply:
I've taken the suggestion to split them into blocks for better SLAP. I prefer to include more detailed messages that explain what the error is specifically rather than a generic one-for-all approach.

Reply:
I was thinking we could use guard clauses here to help with the arrowhead
```
if (!Payment.isValidPaymentFormat(trimmedPayment) &&
    !Payment.isNumberWithAnyDecimals(trimmedPayment)) {
    // If trimmedPayment is not a valid number
    throw new ParseException(Payment.MESSAGE_CONSTRAINTS);
}

if (!Payment.isValidPaymentFormat(trimmedPayment)) {
    // trimmedPayment is a number with incorrect decimal places
    throw new ParseException(Payment.DECIMAL_CONSTRAINTS);
}
            
if (!Payment.isValidPaymentAmount(trimmedPayment)) {
     // Payment amount is greater than maximum allowed
     throw new ParseException(Payment.AMOUNT_CONSTRAINTS);
}
```